Firestore/core/src/firebase/firestore/model/field_value.h

```diff
@@ -38,6 +40,12 @@ struct ServerTimestamp {
   bool has_previous_value_;
 };
 
+struct DatabaseReference {
```

Comment:
This is called a `ReferenceValue` on the other SDKs. Let's keep that name here.

Reply:
It will collide with the name of the function `FieldValue::ReferenceValue(...)` and in that function we actually need to create the hidden structure, which we cannot because it is hidden by the function with the same name.

Reply:
Yes, but the name "DatabaseReference" conveys a different concept entirely. I'd rather apply some kind of general deconfliction strategy than introducing new concepts.

For example, if there's a general problem that these structs need to have the same name as the function I'd prefer we name all of them something like FooContents or FooData. That way the relationship with the canonical concept is readily apparent and the pattern makes it obvious that we're merely deconflicting.

We should be very careful to avoid introducing things that appear to be new concepts (if only the on the surface) just to avoid naming collisions.

Reply:
fixed by using qualified name explicitly.